### PR TITLE
New accordion component

### PR DESCRIPTION
**Pull request by kulmann**
- [x] add documentation blocks for component props
- [x] animation on chevron change (when collapsing/expanding accordion items)

### CODE REVIEW

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398&bid=20767267)

File: src/styles/theme/oc-accordion.scss [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+$accordion-title-font-size:                     $global-medium-font-size !default;
+$accordion-title-line-height:                   1.4 !default;
+$accordion-title-color:                         $global-emphasis-color !default;
+$accordion-title-hover-color:                   $global-color !default;
+
+$accordion-content-margin-top:                  $global-margin !default;
+
+.oc-accordion {
+  padding: 0;
+  list-style: none;
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:12]: Codacy found an issue: [Expected list-style to come before padding (order/properties-alphabetical-order)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 2]
```diff
@@ -0,0 +1,48 @@
+$accordion-item-margin-top:                     $global-margin !default;
+
+$accordion-title-font-size:                     $global-medium-font-size !default;
+$accordion-title-line-height:                   1.4 !default;
+$accordion-title-color:                         $global-emphasis-color !default;
+$accordion-title-hover-color:                   $global-color !default;
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:6]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/elements/OcAccordion.vue [hunk 3]
```diff
@@ -27,30 +27,28 @@ export default {
       return "multiple: " + this.multiple
     },
   },
+  mounted() {
+    this.$on("toggle", id => {
+      const collapseOthers = !this.multiple
```
Comment (micbar) [inline src/elements/OcAccordion.vue:32]: Codacy found an issue: ['collapseOthers' is assigned a value but never used.](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 4]
```diff
@@ -0,0 +1,48 @@
+$accordion-item-margin-top:                     $global-margin !default;
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:1]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/elements/OcAccordion.vue [hunk 5]
```diff
@@ -27,30 +27,28 @@ export default {
...
+      this.$el.children.forEach(child => {
+        const toggled = child.attributes.id.nodeValue === id
```
Comment (micbar) [inline src/elements/OcAccordion.vue:34]: Codacy found an issue: ['toggled' is assigned a value but never used.](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+$accordion-title-font-size:                     $global-medium-font-size !default;
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:3]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+$accordion-title-line-height:                   1.4 !default;
+$accordion-title-color:                         $global-emphasis-color !default;
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:5]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+.oc-accordion > :nth-child(n+2) {
+  margin-top: $accordion-item-margin-top;
+}
+
+.oc-accordion-title {
+  display: block;
+  font-size: $accordion-title-font-size;
+  line-height: $accordion-title-line-height;
+  color: $accordion-title-color;
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:23]: Codacy found an issue: [Expected color to come before line-height (order/properties-alphabetical-order)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 9]
```diff
@@ -0,0 +1,48 @@
+$accordion-item-margin-top:                     $global-margin !default;
+
+$accordion-title-font-size:                     $global-medium-font-size !default;
+$accordion-title-line-height:                   1.4 !default;
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:4]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  outline: none;
+}
+
+.oc-accordion-content {
+  margin-top: $accordion-content-margin-top;
+}
+
+.oc-accordion-content::before,
+.oc-accordion-content::after {
+  content: "";
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:39]: Codacy found an issue: [Expected single quotes (string-quotes)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 11]
```diff
@@ -0,0 +1,48 @@
+$accordion-item-margin-top:                     $global-margin !default;
+
+$accordion-title-font-size:                     $global-medium-font-size !default;
+$accordion-title-line-height:                   1.4 !default;
+$accordion-title-color:                         $global-emphasis-color !default;
+$accordion-title-hover-color:                   $global-color !default;
+
+$accordion-content-margin-top:                  $global-margin !default;
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:8]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+  font-size: $accordion-title-font-size;
+  line-height: $accordion-title-line-height;
+  color: $accordion-title-color;
+}
+
+.oc-accordion-title:hover,
+.oc-accordion-title:focus {
+  color: $accordion-title-hover-color;
+  text-decoration: none;
+  outline: none;
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:30]: Codacy found an issue: [Expected outline to come before text-decoration (order/properties-alphabetical-order)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398&bid=20767267)

File: src/styles/theme/oc-accordion.scss (hunk 10 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:39]: Codacy found an issue: [Expected single quotes (string-quotes)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss (hunk 2 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:6]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss (hunk 9 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:4]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss (hunk 7 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:5]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss (hunk 6 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:3]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss (hunk 1 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:12]: Codacy found an issue: [Expected list-style to come before padding (order/properties-alphabetical-order)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss (hunk 12 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:30]: Codacy found an issue: [Expected outline to come before text-decoration (order/properties-alphabetical-order)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss (hunk 4 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:1]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss (hunk 11 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:8]: Codacy found an issue: [Expected single space after ":" (scss/dollar-variable-colon-space-after)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss (hunk 8 above)
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:23]: Codacy found an issue: [Expected color to come before line-height (order/properties-alphabetical-order)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398&bid=20767267)

File: src/styles/_owncloud.scss [hunk 13]
```diff
@@ -51,3 +51,4 @@
 @import 'theme/oc-modal';
 @import 'theme/oc-sidebar';
 @import 'theme/oc-radio';
+@import 'theme/oc-accordion';
```
Comment (micbar) [inline src/styles/_owncloud.scss:54]: Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398&bid=20767267)

File: src/styles/_owncloud.scss (hunk 13 above)
Comment (micbar) [inline src/styles/_owncloud.scss:54]: Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398&bid=20767267)

File: src/styles/_owncloud.scss (hunk 13 above)
Comment (micbar) [inline src/styles/_owncloud.scss:54]: Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398&bid=20767267)

File: src/styles/_owncloud.scss (hunk 13 above)
Comment (micbar) [inline src/styles/_owncloud.scss:54]: Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398&bid=20767267)

File: src/styles/_owncloud.scss (hunk 13 above)
Comment (micbar) [inline src/styles/_owncloud.scss:54]: Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+      color: $color;
+      outline: none;
+      text-decoration: none;
+
+      .oc-icon {
+        transition-duration: 0.3s;
+        &.rotate {
+          transform: rotate(180deg);
+        }
+        > svg {
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:27]: Codacy found an issue: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

File: src/styles/theme/oc-accordion.scss [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    > .oc-button,
+    > .oc-button:hover,
+    > .oc-button:focus {
+      color: $color;
+      outline: none;
+      text-decoration: none;
+
+      .oc-icon {
+        transition-duration: 0.3s;
+        &.rotate {
```
Comment (micbar) [inline src/styles/theme/oc-accordion.scss:24]: Codacy found an issue: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=6407398)

**Review by LukasHirt — APPROVED** [review]

**Comment by kulmann** [comment]
Follow up issue: https://github.com/owncloud/owncloud-design-system/issues/914

**Pull request merged**